### PR TITLE
Feature/merged home my forest merged develop

### LINKED ISSUE #839: Web: User Home + My Trees Page
OLD :: Design is available at: https://zpl.io/bA7XZAP

New Design: 
https://zpl.io/VK6P6jE
https://zpl.io/Vxrl4Z3

You may want to review http://trilliontreecampaign.org/home 
https://www.trilliontreecampaign.org/my-trees

Recurrent Donations API is not yet available. You can still create UI.

for `All Contributions` there is an API, you can copy implementation from `My Trees` Page

All the contribution cards can be clicked, it will take user to a different page with details about the contribution.

Delete shows up a popup, (there should be preexisting popup component) (implemented in My Trees Page)
Edit should redirect user to new page to edit (currently exists).

### PR DESCRIPTION
**Pull request by mahmed0715**
Fix #839
FIX https://github.com/Plant-for-the-Planet-org/treecounter-app/issues/1700
Updated PR from https://github.com/Plant-for-the-Planet-org/treecounter-app/pull/1202

### CODE REVIEW

File: app/components/UserContributions/RecurringCard.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            <i className="material-icons">delete_outline</i>
+          </span>
+          <span>
+            <i className="material-icons">edit</i>
+          </span>
+        </div>
+      </div>
+    </div>
+  );
+};
```
Comment (norbertschuler) [inline app/components/UserContributions/RecurringCard.js]: Is this code used? I can not image it with all these text resources, dates and numbers in it?

File: app/components/UserContributions/ContributionCard.js [hunk 2]
```diff
@@ -42,68 +45,72 @@ export default class ContributionCard extends React.Component {
...
-        date: formatDate(plantDate)
-      }) +
-      ', ' +
-      i18n.t('label.added_on', {
-        date: formatDate(registrationDate)
-      })
-    );
+  // eslint-disable-next-line no-unused-vars
+  plantActionLine(contribution) {
+    return i18n.t('label.renews_on', {
```
Comment (norbertschuler) [inline app/components/UserContributions/ContributionCard.js]: Does it make sense to write "renewed on" for all these three different types of dates, e.g. plantDate, redemptionDate and registrationDate (which was formerly used with a resource like "added on"?

File: app/components/PublicTreeCounter/TreecounterHeader.js [hunk 3]
```diff
@@ -15,11 +15,16 @@ const TreecounterHeader = ({
...
+    let c = text.split(' ');
+    if (c.length > 1) c.length--;
+    return c.join(' ');
+  };
   return (
     <div className="tree-counter-profile flex-column">
       <UserProfileImage profileImage={logo} />
       <div className="user-info">
-        <div className="tree-counter-name">{caption}</div>
+        <div className="tree-counter-name">{onlyFirstName(caption)}</div>
```
Comment (norbertschuler) [inline app/components/PublicTreeCounter/TreecounterHeader.js]: Sagar said, we should use the fullname: https://github.com/Plant-for-the-Planet-org/treecounter-app/pull/1202#discussion_r312424538

**Comment by norbertschuler** [comment]
I closed the old PR and copied the issues into this one. I am not sure if you already fixed the stuff I mentioned in this comment of the old PR: https://github.com/Plant-for-the-Planet-org/treecounter-app/pull/1202#issuecomment-542629715

File: app/components/UserContributions/RecurringCard.js (hunk 1 above)
Comment (mahmed0715) [reply app/components/UserContributions/RecurringCard.js]: This code is not being used for now, The base deign had this thus we have it here, I will remove this from here.

File: app/components/UserContributions/ContributionCard.js (hunk 2 above)
Comment (mahmed0715) [reply app/components/UserContributions/ContributionCard.js]: The renews on is not being used now, but later we will show something like that. changed to something general.. its only date for now ``` "renews_on": "{{date}}``` .
though changed.

File: app/components/PublicTreeCounter/TreecounterHeader.js (hunk 3 above)
Comment (mahmed0715) [reply app/components/PublicTreeCounter/TreecounterHeader.js]: true. changed

**Comment by mahmed0715** [comment]
Yes @norbertschuler I have fixed all the reported issues in old PR by you.

File: app/components/UserContributions/ContributionCard.js [hunk 4]
```diff
@@ -42,68 +45,72 @@ export default class ContributionCard extends React.Component {
...
-        date: formatDate(plantDate)
-      }) +
-      ', ' +
-      i18n.t('label.added_on', {
-        date: formatDate(registrationDate)
-      })
-    );
+  // eslint-disable-next-line no-unused-vars
+  plantActionLine(contribution) {
+    return i18n.t('label.date_on', {
```
Comment (norbertschuler) [inline app/components/UserContributions/ContributionCard.js]: Without any text within the translation resource we also could remove this resource from the resource files and just return the date string here.

**Comment by norbertschuler** [comment]
@sagararyal please check (best of staging backend) if data is still displays correctly as before.

**Comment by mahmed0715** [comment]
@sagararyal I tried to update the native tree contribution card. Please see...

**Comment by norbertschuler** [comment]
This pull request could need another merge of the develop branch and resolving of the merge conflicts.

**Comment by mahmed0715** [comment]
@sagararyal from my end its done, you can test this PR, if any issues I can solve.
Fixed some issues on mobile as well related to contribution card

**Comment by mahmed0715** [comment]
@norbertschuler will do a merge asap

**Comment by norbertschuler** [comment]
> @norbertschuler will do a merge asap

Last merge probably did not work as Git has not recognized your commit as a merge. I suggest you reset the branch to 71b36888a9e0ed182d0e9e49e960980413f8437a and try again and push force you branch afterwards.

**Comment by mahmed0715** [comment]
@norbertschuler merged develop, @sagararyal lets discuss together if everything is there or we need more changes in it. I think all main changes requested are there in this PR.

**Comment by mahmed0715** [comment]
Screenshot from 2019-12-02 21-53-15

Screenshot from 2019-12-02 21-53-24
Screenshot from 2019-12-02 21-53-33
Screenshot from 2019-12-02 21-53-45

@sagararyal here how it looks on web

**Comment by mahmed0715** [comment]
Screenshot from 2019-12-02 22-17-25
after update giver avatar

File: .gitignore [hunk 5]
```diff
@@ -46,4 +46,5 @@ package-lock.json
 web/prod
 web/server
 app/config/index.js
+/nbproject/private/
```
Comment (norbertschuler) [inline .gitignore]: I am curious - what tool creates this folders?

File: .eslintrc.json [hunk 6]
```diff
@@ -42,7 +42,7 @@
     "no-undef": 2,
     "no-underscore-dangle": 2,
     "no-unreachable": 2,
-    "no-unused-vars": 2,
+    "no-unused-vars": 1,
```
Comment (norbertschuler) [inline .eslintrc.json]: Why do we lower the level for `no-unused-vars` again?

File: .gitignore (hunk 5 above)
Comment (mahmed0715) [reply .gitignore]: ah, once I installed netbeans ide, thats weird. that created this folder

File: .eslintrc.json (hunk 6 above)
Comment (mahmed0715) [reply .eslintrc.json]: will revert back, it was done when merging may be..

**Comment by norbertschuler** [comment]
I compared my user contributions on the test backend and marked the differences between the former list and the new one - I do not know what's an error and what is made on purpose - @mahmed0715 @sagararyal do you have checked and discussed all this?

Bildschirmfoto 2019-12-05 um 16 02 03
Bildschirmfoto 2019-12-05 um 16 02 28

**Comment by norbertschuler** [comment]
I also compare by user contributions on staging backend now and found more stuff which is now not shown any more. Also on staging not all user contribution can be shown as the layout does not allow it (I tweak my browser to make this screen shots):

Bildschirmfoto 2019-12-05 um 16 25 21
Bildschirmfoto 2019-12-05 um 16 22 09
Bildschirmfoto 2019-12-05 um 16 23 59

**Comment by norbertschuler** [comment]
This is how staging looks to me originally (no map, no titles or text above my contributions):
Bildschirmfoto 2019-12-05 um 16 31 51

**Comment by mahmed0715** [comment]
> This is how staging looks to me originally (no map, no titles or text above my contributions):
> Bildschirmfoto 2019-12-05 um 16 31 51

You may checked my trees on this PR, we are not managing my-trees anymore, instead we moved to /home that is for now accessible by clicking on the top left image(Plant for the planel image)

File: .gitignore (hunk 5 above)
Comment (norbertschuler) [reply .gitignore]: If you work this that IDE, keep this ignore. We don't want to commit these files :-)

**Comment by norbertschuler** [comment]
> You may checked my trees on this PR, we are not managing my-trees anymore, instead we moved to /home that is for now accessible by clicking on the top left image(Plant for the planel image)

Oh, that's my fault, sorry. I added the right screen shot above.

File: app/components/UserContributions/RecurringCard.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import * as images from '../../assets';
+
+// eslint-disable-next-line no-unused-vars
+const RecurringCard = ({ cardData, deleteContribution }) => {
```
Comment (norbertschuler) [inline app/components/UserContributions/RecurringCard.js]: The component `RecurringCard` is unused and the containing resources are not extracted. We should delete this file as well as the file `app/components/UserContributions/Contributors.js` which is also unused and empty.

Comment (norbertschuler) [inline app/components/UserContributions/RecurringCard.js]: The component `RecurringCard` is unused and the containing resources are not extracted. We should delete this file as well as the file `app/components/UserContributions/Contributors.js` which is also unused and empty.

Comment (norbertschuler) [inline app/components/UserContributions/RecurringCard.js]: The component `RecurringCard` is unused and the containing resources are not extracted. We should delete this file as well as the file `app/components/UserContributions/Contributors.js` which is also unused and empty.

**Pull request closed without merging**